lib/charms/mongodb/v0/mongodb_backups.py

```diff
@@ -515,16 +540,50 @@ def _try_to_restore(self, backup_id: str) -> None:
                         error_message = e.output.decode("utf-8")
                     else:
                         error_message = str(e.stderr)
-                    fail_message = f"Restore failed: {str(e)}"
-
                     if "Resync" in error_message:
                         raise ResyncError
 
+                    fail_message = f"Restore failed: {str(e)}"
                     if f"backup '{backup_id}' not found" in error_message:
                         fail_message = f"Restore failed: Backup id '{backup_id}' does not exist in list of backups, please check list-backups for the available backup_ids."
 
                     raise RestoreError(fail_message)
 
+    def _try_to_backup(self):
+        """Try to create a backup and return the backup id.
+
+        If PBM is resyncing, the function will retry to create backup
+        (up to BACKUP_RESTORE_MAX_ATTEMPTS times)
+        with BACKUP_RESTORE_ATTEMPT_COOLDOWN time between attepts.
+
+        If PMB returen any other error, the function will raise BackupError.
+        """
```

Comment:
**out-of-scope** I believe you have great doc strings. One possible improvement may be to provides the information about "Args:" , "Returns:" and "Raises:", see e.g. [here](https://github.com/canonical/upload-charm-docs/blob/main/src/repository.py#L105)

Of course, one could start bit by bit, or do a refactoring all together. Really up to your team to whether this should be followed up and how. Also, private/hidden methods are not required to have kick-ass doc strings, I would start from the public ones